puzzle.js

```diff
@@ -790,20 +801,24 @@ function PuzzleEntry(p) {
             if (clueNumbers && shape[r][c] != '@') {
                 var acrossClue = (c == 0 || shape[r][c-1] == '@' || (edgeCode & 4)) && c < shape[r].length - 1 && shape[r][c+1] != '@' && !(edgeCode & 8); // block/edge left, letter right
                 var downClue = (r == 0 || shape[r-1][c] == '@' || (edgeCode & 1)) && r < shape.length - 1 && shape[r+1][c] != '@' && !(edgeCode & 2); // block/edge above, letter below
+                
                 if (acrossClue || downClue) {
                     var clueRealNum = (clueNumbers == "auto") ? ++clueNum : clueNumbers[clueNum++];
-
-                    if (acrossClue) { td.setAttribute("data-across-cluenumber", clueRealNum); }
-                    if (downClue) { td.setAttribute("data-down-cluenumber", clueRealNum); }
-                    if (acrossClue && acrossClues[acrossClueIndex]) {
-                      acrossClues[acrossClueIndex].setAttribute("data-across-cluenumber", clueRealNum);
-                      acrossClues[acrossClueIndex].setAttribute("value", clueRealNum);
-                      acrossClueIndex++;
-                    }
-                    if (downClue && downClues[downClueIndex]) {
-                      downClues[downClueIndex].setAttribute("data-down-cluenumber", clueRealNum);
-                      downClues[downClueIndex].setAttribute("value", clueRealNum);
-                      downClueIndex++;
+                    const isClueEmpty = `${clueRealNum}`.trim() === "";
```

Comment:
is this equivalent to clueRealNum.toString()? If so I'd prefer the toString form since nothing complex is happening. The current format gives me jQuery vibes, lol